packages/embed-widget/src/App.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import {
+  ContextMenuRoot,
+  ErrorBoundary,
+  LoadingOverlay,
+} from '@deephaven/components'; // Use the loading spinner from the Deephaven components package
+import type { VariableDefinition } from '@deephaven/jsapi-types';
+import Log from '@deephaven/log';
+import './App.scss'; // Styles for in this app
+import { useConnection } from '@deephaven/jsapi-components';
+import { fetchVariableDefinition } from '@deephaven/jsapi-utils';
+import { WidgetView } from '@deephaven/plugin';
+
+const log = Log.module('EmbedWidget.App');
+
+/**
+ * A functional React component that displays a Deephaven Widget using the @deephaven/plugin package.
+ * It will attempt to open and display the widget specified with the `name` parameter, expecting it to be present on the server.
+ * E.g. http://localhost:4030/?name=myWidget will attempt to open a widget `myWidget`
```

Comment:
Yea it probably would never be _too_ expensive, but we certainly can add `type` as an optional param.

Reply:
Actually, we can't really save anything here - when we call getObject with just the name/type defined, it still looks up the variable definition object: https://github.com/deephaven/deephaven-core/blob/374188ae166aa202da30e078a78c13334cb683f6/web/client-api/src/main/java/io/deephaven/web/client/api/WorkerConnection.java#L803
If we pass it the instance of the variable definition, we skip up looking up that variable definition: https://github.com/deephaven/deephaven-core/blob/374188ae166aa202da30e078a78c13334cb683f6/web/client-api/src/main/java/io/deephaven/web/client/api/WorkerConnection.java#L789
So providing `type` can basically just get you in a pickle, if it doesn't match what's on the server, and it's not saving us anything since we need to get the definition anyway.